internal/backends/backend_test.go

```diff
@@ -64,3 +64,71 @@ func TestVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestListDatabases(t *testing.T) {
+	t.Parallel()
+
+	ctx := conninfo.Ctx(testutil.Ctx(t), conninfo.New())
+
+	for name, b := range testBackends(t) {
+		name, b := name, b
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			// setup 3 DB with 1 collections each. random order of db name also ensure sorting test.
+			dbNames := []string{"testDB2", "testDB1", "testDB3"}
+			collectionName := "testCollection"
+
+			testDB, err := b.Database(dbNames[0])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionName})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[0]}) //nolint:errcheck // safe to ignore
```

Comment:
That's not needed because the whole PostgreSQL database / SQLite directory is removed at the end of the test

Reply:
Done.